src/state/hooks/DatasetHooks.js

```diff
@@ -29,7 +31,10 @@ export const useAddDatasetToStore = () => {
 export const useCurrentDataset = () => {
   const datasets = useDatasetListData();
   const selectedDatasetIndex = useSelectedDatasetIndex();
-  return datasets?.[selectedDatasetIndex] ?? null;
+  return (
+    datasets?.[selectedDatasetIndex] ??
+    ResourceUtils.getResourceTree(datasets?.filter((dataset) => dataset.main === true))[0]
```

Comment:
Yes, we can, it's never too safe :laughing: but I don't think it will raise an error because the components won't render if main list is empty (the placeholder is rendered instead), in my tests of deleting the last dataset, the placeholder was shown as expected. However, it can be useful to have a fallback :+1:

Reply:
Yeah, I fear that the extra safeguard may still be useful, because we don't know where this "generic" hook will be used in the future :sweat_smile: